src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -1374,6 +1374,42 @@ public void ICollectionSyncRoot_NotSupported()
             ICollection c = ImmutableArray.Create(1, 2, 3);
             Assert.Throws<NotSupportedException>(() => c.SyncRoot);
         }
+        
+        //[Fact] Perf test
+        public void InsertRange_Perf_Arrays()
+        {
+            TestInsertRangePerfWith(Enumerable.Range(1, 10000).ToArray());
+        }
+        
+        //[Fact] Perf test
+        public void InsertRange_Perf_ListOfT()
+        {
+            TestInsertRangePerfWith(Enumerable.Range(1, 10000).ToList());
+        }
+        
+        //[Fact] Perf test
+        public void InsertRange_Perf_NonCollectionEnumerable()
+        {
+            // IEnumerable being tested here must not inherit from
+            // ICollection<T>, ICollection, or IReadOnlyCollection<T>.
+            
+            TestInsertRangePerfWith(Enumerable.Range(1, 10000));
+        }
+        
+        private void TestInsertRangePerfWith(IEnumerable<T> toInsert)
```

Comment:
Can this be made `static`?
Also (stylecop) please move the private methods below the protected methods.